aioshelly/rpc_device/wsrpc.py

```diff
@@ -394,13 +397,14 @@ async def _rpc_call(
         if self._client is None:
             raise RuntimeError("Not connected")
 
-        call = RPCCall(self._next_id, method, params, self._session)
+        future: asyncio.Future[dict[str, Any]] = self._loop.create_future()
+        call = RPCCall(self._next_id, method, params, self._session, future)
         self._calls[call.call_id] = call
 
         try:
             async with async_timeout.timeout(timeout):
                 await self._send_json(call.request_frame)
-                resp: dict[str, Any] = await call.resolve
+                resp = await future
         except asyncio.TimeoutError as exc:
             with contextlib.suppress(asyncio.CancelledError):
                 call.resolve.cancel()
```

Comment:
Should this also be changed to cancel and await `future`?

Reply:
It still works either way, but cleaner for sure

Reply:
will adjust shortly

Reply:
d834485f833336137d86c4625b495742a4af5752